src/components/Profile/ProfileInfoCard/editProfileDetailsModal.js

```diff
@@ -59,162 +95,319 @@ const EditProfileDetailsModal = ({ profileData, modelCloseCallback }) => {
 
   useEffect(() => {
     if (loading === false && error === false) {
-      message.success("Changes saved!");
       closeModal();
     }
   }, [closeModal, loading, error]);
 
-  const getData = prop => (Boolean(prop) ? prop : "");
-
-  useEffect(() => {
-    form.setFieldsValue({
-      displayName: getData(profileData.displayName),
-      website: getData(profileData.website),
-      link_facebook: getData(profileData.link_facebook),
-      link_github: getData(profileData.link_github),
-      link_linkedin: getData(profileData.link_linkedin),
-      link_twitter: getData(profileData.link_twitter),
-      description: getData(profileData.description),
-      org_country: getData(profileData.country)
-    });
-  }, [form, profileData]);
-
-  const onSubmit = formData => {
-    const {
-      displayName,
-      website,
-      link_facebook,
-      link_github,
-      link_linkedin,
-      link_twitter,
+  const validated = () => {
+    const countryValid = validateCountry(country, setCountryValidateError);
+    const orgWebsiteValid = validateOrgWebsite(website, setWebsiteValidateError, setWebsiteValidateErrorMessage);
+    const nameValid = validateName(
+      name,
+      setNameValidateError,
+      setNameValidateErrorMessage,
+      "Please enter your name",
+      "Please enter a real name"
+    );
+    const descriptionValid = validateIsEmpty(
       description,
-      org_country: country
-    } = formData;
-    updateUserProfile({
-      displayName,
-      website,
-      link_facebook,
-      link_github,
-      link_linkedin,
-      link_twitter,
-      description,
-      country
-    })(firebase, firestore, dispatch);
+      setDescriptionValidateError,
+      setDescriptionValidateErrorMessage,
+      "Please enter a description"
+    );
+    const facebookValid = validateIsEmpty(
+      facebook,
+      setFacebookValidateError,
+      setFacebookValidateErrorMessage,
+      "Please enter a Facebook username"
+    );
+    const twitterValid = validateIsEmpty(
+      twitter,
+      setTwitterValidateError,
+      setTwitterValidateErrorMessage,
+      "Please enter a Twitter username"
+    );
+    const linkedinValid = validateIsEmpty(
+      linkedin,
+      setLinkedinValidateError,
```

Comment:
Can't we move these into that separate helper file we had earlier for validations? If it is not doable, that's fine. But would be great if it is possible.

Reply:
Actually they are imported from a helper file that I previously wrote.
`import { validateName, validateCountry, validateOrgWebsite, validateIsEmpty } from "../../../helpers/validations";`
**ValidateIsEmpty** is a function I wrote. I called them using different strings as parameters just to avoid using a lot of functions.

Reply:
Oh, I see. My bad then.